### PR TITLE
Adding ability to supply an HTTP Agent

### PR DESCRIPTION
**Pull request by seangallavan**
This is helpful if you want to access vault behind a bastion.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -24,12 +24,21 @@ module.exports = (config = {}) => {
   tv4 = config.tv4 || tv4;
   commands = config.commands || commands;
   mustache = config.mustache || mustache;
-  rp = (config['request-promise'] || rp).defaults({
+
+  let rpDefaults = {
```
Comment (petercunha) [inline src/index.js]: Please change this to a const. The linter is causing the travis build to fail.

**Comment by seangallavan** [comment]
Updated to latest master...believe it's good to go @petercunha

**Comment by petercunha** [comment]
Looks good, hopefully one of the maintainers will review it soon 👍

**Comment by kr1sp1n** [comment]
@seangallavan Thanks for your PR and sorry for the very late merge.
@petercunha Released as v0.9.17 https://www.npmjs.com/package/node-vault/v/0.9.17

**Pull request merged**